### ISSUE TITLE
MapView from React-Native-Maps appears as a grey block on Android

### ISSUE THREAD

**Issue by Mikkal24** [expo/expo#18554]
Edit from Expo, [see the latest status of this issue here](https://github.com/expo/expo/issues/18554#issuecomment-1216561781).

---

### Summary

When running any app following the instructions on https://docs.expo.dev/versions/latest/sdk/map-view the map appears as a gray block on android. *NOTE* iOS seems to be working fine?

<img src="https://user-images.githubusercontent.com/21202515/183310225-383d7f70-b6f2-476d-9c39-83916600cb69.png" width="100" />

### Managed or bare workflow? If you have made manual changes inside of the `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### Package versions

```  
"dependencies": {
    "expo": "^46.0.0",
    "expo-status-bar": "~1.4.0",
    "react": "18.0.0",
    "react-dom": "18.0.0",
    "react-native": "0.69.3",
    "react-native-web": "~0.18.7",
    "react-redux": "^8.0.2",
    "react-native-maps": "0.31.1"
  },
```

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 5.18 Pop!_OS 21.10
      Shell: 5.1.8 - /bin/bash
    Binaries:
      Node: 16.13.1 - /usr/local/lib/node/nodejs/bin/node
      npm: 8.1.2 - /usr/local/lib/node/nodejs/bin/npm
    npmPackages:
      expo: ^46.0.0 => 46.0.2 
      react: 18.0.0 => 18.0.0 
      react-dom: 18.0.0 => 18.0.0 
      react-native: 0.69.3 => 0.69.3 
      react-native-web: ~0.18.7 => 0.18.7 
    npmGlobalPackages:
      expo-cli: 6.0.1
    Expo Workflow: managed
```

### Reproducible demo

https://snack.expo.dev/xdxj1hUb7

### Stacktrace (if a crash is involved)

N/A

**Comment by BrodaNoel** [comment]
Have you added your Google Maps API Key in `app.json`?

You need this inside your "android" key:
```
"config": {
  "googleMaps": {
    "apiKey": "xxxxx"
  }
},
```

**Comment by Mikkal24** [comment]
> Have you added your Google Maps API Key in `app.json`?
> 
> You need this inside your "android" key:
> 
> ```
> "config": {
>   "googleMaps": {
>     "apiKey": "xxxxx"
>   }
> },
> ```

I did yeah since I encountered that solution on [GH](https://github.com/expo/expo/issues/5591) but that did not resolve the issue. 

Were you able to get it to work using Expo go and an API key?

Also, the docs mention that no additional setup should be necessary when using Expo Go. 

> No additional configuration is necessary to use react-native-maps in Expo Go. However, once you want to deploy your standalone app you should follow instructions below.

https://docs.expo.dev/versions/latest/sdk/map-view/

**Comment by iM-GeeKy** [comment]
Granted there isn't a misconfiguration in your config file, it could potentially be related to https://github.com/expo/expo/issues/18530. I never confirmed what happens on android emulator, but Google Maps did seem to render on a physical Android device following the documentation.

**Comment by Mikkal24** [comment]
its hard for me to speculate since I don't know how to look at logs for this issue but, coming from web development and dealing with API key misconfigurations it does smell like an API key issue. Though I have absolutely no idea how it is configured in expo. 

That being said, the iOS version of the [snack demo I provided](https://snack.expo.dev/xdxj1hUb7?platform=ios) does seem to work. It just shows up as gray on android. Very strange.

**Comment by forki** [comment]
we have a similar situation. the expo go app doesn't show the map anymore - BUT if we release it, then the real production app still shows the map. So I assume something in expo go is wrong

**Comment by Mikkal24** [comment]
So, I spent the time figuring out how to launch the app with the bare workflow. And surprisingly it works even without an API key. 

Very strange but, at least there's that.

**Comment by Mikkal24** [comment]
This issue really stuck in my brain for whatever reason. It does look like the issue is that the API Key is not being sent to the manifest file from app.json for whatever reason. 

So I made a plugin file which should work out of the box with the managed workflow

```js
const { withAndroidManifest } = require("@expo/config-plugins")
module.exports = function androidManifestPlugin(config) {
    return withAndroidManifest(config, async config => {

        let androidManifest = config.modResults.manifest


        const googleManifest = androidManifest.application.find(meta => meta.$["android:name"] === "com.google.android.geo.API_KEY");
        if (googleManifest) {
            googleManifest.$["android:value"] = config.android.config.googleMaps.apiKey
        } else {
            androidManifest.application.push({
                $: {
                    "android:name": "com.google.android.geo.API_KEY",
                    "android:value": config.android.config.googleMaps.apiKey
                }
            })
        }


        return config
    })
}
```

basically just add this somewhere in your project and reference it in your `app.json` file

e.g.

```json
{
  "expo": {
    "plugins": [
      "./google-maps.plugin"
    ],
    "android": {
      "config": {
        "googleMaps": {
          "apiKey": "{your api key here}"
        }
      },
    .
    .
    .
}
```

**Comment by forki** [comment]
My assumption is that this issue and https://github.com/expo/expo/issues/18570 are related. In both cases it seems the expo go app does not have access to the properties that are provided through the enviroment like app.json

**Comment by da45** [comment]
a serious and blocking issue, should I turn back to SDK 45 ?! why no one assigned to this bug?!!

**Comment by Mikkal24** [comment]
> a serious and blocking issue, we can't go further in our project! , should we turn back to SDK 45 ?! why no one assigned to this bug?!!

Have you tried converting your project to a bare workflow? That should allow you to add your google maps API key manually in the android files and allow you to keep building while we wait for a resolution to the managed version.

**Comment by brentvatne** [comment]
you can create a development build of your app with your own api key, without needing to use 'bare workflow': https://docs.expo.dev/development/introduction/

**Comment by forki** [comment]
@brentvatne development builds don't allow hot reload, right? We use expo go during dev in Wi-Fi because unlike an emulator it gives all features of a real device.

**Comment by Mikkal24** [comment]
Based on your suggestion I moved to the expo development build rather than the bare workflow. And the map showed up right away no API key necessary. 

However, I continued to plug along and move my firebase implementation to the [legit version of firebase](https://rnfirebase.io/) rather than using the JS SDK that I was using with expo go. 

And the map immediately started crashing lol. Like it started expecting an API key when firebase got added. 

I've moved to using mapbox instead while I wait for this issue to be addressed but, man this development experience has been rough to say the least.

**Comment by da45** [comment]
> you can create a development build of your app with your own api key, without needing to use 'bare workflow': https://docs.expo.dev/development/introduction/

forget about my case, why it does not work in https://snack.expo.dev ? do you think you gave the best answer by asking to create dev build !!! there is a breaking change , it worded in SDK 45 and previous versions, so it must work in 46 too, don't ask  users to go with other alternatives !!!

**Comment by forki** [comment]
@da45 please calm down. It's good engineering practice to try to find workarounds. Often it unblocks at least some percentage of users.

**Comment by BrodaNoel** [comment]
@da45 I agree it is a breaking change and it is important. So why don't you fix it? Or why didn't you help to test it before the release?

It's an open source library, so, if you don't fix it, or if you didn't help to test it, consider yourself as the guilty that this error still exists.

**Comment by BrodaNoel** [comment]
And if it is working on SDK45, why in the heaven don't you downgrade to SDK45 until this gets fixed?!

**Comment by hlthi** [comment]
It is not SDK 46  problem! @brentvatne @da45 

With SDK 45,  Expo Go 2.25.1 grey block on android, crash on ios. 
With SDK 46, Expo GO 2.25.1 grey block on android, crash on ios. 

But,
With SDK 45, Expo Go 2.24.3 work.

How to downgrade Expo 45 with  Expo Go 2.24.3.

```
- Delete Expo Go app from Simulator and Real Device.
1. expo upgrade 45
2. expo start
  - Expo , automatically install expo go 2.24.3.
```
Now you can wait for fix Expo Go problem :)

**Comment by Pitouli** [comment]
@hlthi 
Hello, how can I reinstall Expo Go 2.24.3 ?
On my phone, I have Expo Go 2.25.1 but I do not know how to downgrade it ^^'

**Comment by EquesMedia** [comment]
> you can create a development build of your app with your own api key, without needing to use 'bare workflow': https://docs.expo.dev/development/introduction/

Hi @brentvatne, I tried with a custom development build with `SDK 46` and `react-native-maps 0.31.1` and the map is blank. I am not using Expo Go. API key is loaded via `app.config.js` on `android.config.googleMaps.apiKey` as indicated in the expo documentation.

Did you make it work on your end?

**Comment by BrodaNoel** [comment]
I am using 'app.json' and I don't have this problem. Are you all using 'app.config.js'?

**Comment by EquesMedia** [comment]
> I am using 'app.json' and I don't have this problem. Are you all using 'app.config.js'?

`app.config.js` is just a dynamic version of `app.json` that I use for app variants. Fmi: https://docs.expo.dev/workflow/configuration/

@BrodaNoel are you using the same version as I am? SDK 46 / react-native-maps 0.31.1 / development build and your API key set on android.config.googleMaps.apiKey in app.json?

**Comment by byCedric** [comment]
As for some of your questions/comments:

---

> we have a similar situation. the expo go app doesn't show the map anymore - BUT if we release it, then the real production app still shows the map. So I assume something in expo go is wrong

Yes, that assumption is right! We found an issue that only relates to the Expo Go builds and not your production apps. Even apps built with the classic build system (`expo build`) are not affected. But, because we use Expo Go in Snack too, Snack is also affected.

---

> @brentvatne development builds don't allow hot reload, right? We use expo go during dev in Wi-Fi because unlike an emulator it gives all features of a real device.

Expo dev client builds are fully capable of doing hot reload. I would suggest to try it out :)

---

> Based on your suggestion I moved to the expo development build rather than the bare workflow. And the map showed up right away no API key necessary.
>
> However, I continued to plug along and move my firebase implementation to the [legit version of firebase](https://rnfirebase.io/) rather than using the JS SDK that I was using with expo go.
>
> And the map immediately started crashing lol. Like it started expecting an API key when firebase got added.
>
> I've moved to using mapbox instead while I wait for this issue to be addressed
>
> [edit] removed my salty comments. I just picked a bad week to pick up react native lol

The Expo dev client and EAS allow you to pick any native module as you see fit. I'm sorry you had an unfortunate experience with `react-native-firebase`, I'm not sure if [you followed these steps](https://rnfirebase.io/#expo) but some libraries require you to add credentials, even in development mode. We can't control that outside of Expo Go.

(And thanks for keeping things civilized ❤️)

---

> forget about my case, why it does not work in https://snack.expo.dev/ ? do you think you gave the best answer by asking to create dev build !!! there is a breaking change , it worded in SDK 45 and previous versions, so it must work in 46 too, don't ask users to go with other alternatives !!!

Because Snack also uses Expo Go, Snack is also affected. We will deploy a fix shortly.

---

> @da45 please calm down. It's good engineering practice to try to find workarounds. Often it unblocks at least some percentage of users.

I fully agree with this. We don't intend to break things, but if they do, we try to keep everyone as "unblocked" as possible while we fix things. We still have to go through store review whenever we fix it, [which took ±15 days on the Play Store for SDK 46](https://twitter.com/cedricvanputten/status/1557375168955252738). So I think there is a lot of value in sharing these workarounds.

---

> Hi @brentvatne, I tried with a custom development build with `SDK 46` and `react-native-maps 0.31.1` and the map is blank. I am not using Expo Go. API key is loaded via `app.config.js` on `android.config.googleMaps.apiKey` as indicated in the expo documentation.
>
> Did you make it work on your end?

If you added the `android.config.googleMaps.apiKey` after running `expo run:*` or `expo prebuild`, it needs to recreate the native files. These keys are set in the native files that `expo prebuild

**Comment by byCedric** [comment]
This should be resolved in the latest Expo Go version! (`2.25.2`)
